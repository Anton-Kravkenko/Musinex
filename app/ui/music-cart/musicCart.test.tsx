import MusicCart from '@/ui/music-cart/musicCart'
import { fireEvent, render } from '@testing-library/react-native'

const mockFn = jest.fn()
describe('musicCart', () => {
	test('should render currect', () => {
		const musicCartRender = render(
			<MusicCart
				text1='search'
				onPress={mockFn}
				wrapClassNames='mb-5'
				text2='artists'
				textCenter
				image={{
					width: 20,
					height: 20,
					border: 10,
					url: 'https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png'
				}}
				style={{
					width: 20
				}}
				className='mb-5'
			/>
		)
		expect(musicCartRender).toMatchSnapshot()
	})
	test('press', () => {
		const { getByTestId } = render(
			<MusicCart
				text1='search'
				onPress={mockFn}
				wrapClassNames='mb-5'
				text2='artists'
				textCenter
				image={{
					width: 20,
					height: 20,
					border: 10,
					url: 'https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png'
				}}
				style={{
					width: 20
				}}
				className='mb-5'
			/>
		)
		fireEvent.press(getByTestId('music-cart-wrapper'))
		expect(mockFn).toHaveBeenCalledTimes(1)
	})
})
