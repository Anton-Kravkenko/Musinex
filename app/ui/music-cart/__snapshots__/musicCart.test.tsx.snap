// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`musicCart should render currect 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  className="mb-5"
  collapsable={false}
  focusable={true}
  image={
    {
      "border": 10,
      "height": 20,
      "url": "https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png",
      "width": 20,
    }
  }
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "maxWidth": 20,
        "width": 20,
      },
      {
        "transform": [
          {
            "scale": 1,
          },
        ],
      },
    ]
  }
  testID="music-cart-wrapper"
  text1="search"
  text2="artists"
>
  <View>
    <Image
      className="mb-5"
      onLoadEnd={[Function]}
      source={
        {
          "height": 20,
          "uri": "https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png",
          "width": 20,
        }
      }
      style={
        [
          {
            "borderRadius": 10,
            "display": "none",
            "height": 20,
            "width": 20,
          },
          {
            "width": 20,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "#1B2021",
            "borderRadius": 6,
            "height": 20,
            "width": 20,
          },
          [
            {
              "borderRadius": 10,
            },
            {
              "width": 20,
            },
          ],
        ]
      }
      testID="skeleton"
    />
  </View>
  <View
    className="mt-2"
  >
    <View
      style={
        [
          {
            "backgroundColor": "#1B2021",
            "borderRadius": 6,
            "height": 16,
            "width": 60,
          },
          undefined,
        ]
      }
      testID="skeleton"
    />
    <View
      style={
        [
          {
            "backgroundColor": "#1B2021",
            "borderRadius": 6,
            "height": 16,
            "width": 70,
          },
          undefined,
        ]
      }
      testID="skeleton"
    />
  </View>
</View>
`;
